WcaOnRails/app/models/user.rb

```diff
@@ -462,7 +462,7 @@ def can_update_crash_course?
   end
 
   def can_manage_competition?(competition)
-    can_admin_results? || competition.organizers.include?(self) || competition.delegates.include?(self) || wrc_team?
+    can_admin_results? || competition.organizers.include?(self) || competition.delegates.include?(self) || wrc_team? || competition.delegates.map { |d| d.senior_delegate }.compact.uniq.include?(self)
```

Comment:
Can we use the &: syntax here? I'm thinking `...delegates.map(&:senior_delegate)...`

I'm guessing the `compact` is to deal with delegates who do not have a senior delegate?

Why are we calling `uniq`?

Reply:
Ah, sorry, I copied from the mailer thing, and it used senior_delegate.email so it needed the { ...} structure.

I don't know about compact, I just used it from there. Is there any delegate with no senior? I don't think so.

uniq also came from the mailer, and I guess it was to prevent having the senior's email twice in the "to" section. Not sure if it's necessary here, since we just want to find out if the current user is in some list.

Reply:
Senior delegates are delegates that don't have senior delegates, so let's definitely keep the `compact`. I do think we can remove the `{...}` and the `uniq`, though.